### PR TITLE
Add @esigo as approver

### PR DESCRIPTION
**Pull request by lalitb**
@esigo has been actively contributing to OpenTelemetry C++ across many modules as well as general improvements to the CI and infrastructure. He has also expressed the willingness to make more contributions in the future.

Few of the recent PRs @esigo has authored ( Total - 40 closed, 2 open)

* #1174
* #1167
* #1164
* #1162
* #1161
* #1155
* #1153
* #1151
* #1150
* #1149
* #1148
* #1144
* #1133
* #1128
* #1123

PRs @esigo has contributed as a reviewer:

* #1178
* #1177
* #1172
* #1171
* #1160
* #1140
* #1137

### CODE REVIEW

**Review by ThomsonTan — APPROVED** [review]

**Comment by reyang** [comment]
+1 Welcome @esigo! 🚀🥳

File: README.md [hunk 1]
```diff
@@ -102,6 +102,7 @@ Approvers
 * [Josh Suereth](https://github.com/jsuereth), Google
 * [Max Golovanov](https://github.com/maxgolov), Microsoft
 * [Ryan Burn](https://github.com/rnburn), Lightstep
+* [Ehsan Saei](https://github.com/esigo)
```
Comment (reyang) [inline README.md:105]: Sort alphabetically unless there is a good reason not to do so.

**Review by reyang — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (lalitb) [reply README.md:105]: No, it should be alphabetic. Will fix.

**Review by esigo — APPROVED** [review]
Thanks for having me. It's a great pleasure to work with you on this project. I'm looking forward to future contributions and collaborations with you :).

**Pull request merged**